### PR TITLE
Maya: fix crashing model renderset collector

### PR DESCRIPTION
**Pull request by antirotor**
## Changelog Description
This fix is handling case where model is in some type of render sets but no other connections are made there. Publishing this model would fail with `RuntimeError: Found no items to list the history for.`

## Testing notes:
1. Create simple model
2. Put it for example into `vrayDisplacement` set
3. Publish

### CODE REVIEW

**Review by LiborBatek — APPROVED** [review]
Works as supposed to. All good!

**Comment by LiborBatek** [comment]
One small note: even tho it works the ```vrayDisplacement``` set is not propagated when publishing ```model``` family.

Is there any OP Settings which allowing it or simply not supported at this time?

**Comment by LiborBatek** [comment]
@antirotor  we should keep the ```vray sets``` when publishing the asset. Should we treat it in separate PR? Because right now it got stripped off. opinions?

**Comment by BigRoy** [comment]
>  we should keep the vray sets when publishing the asset. Should we treat it in separate PR? Because right now it got stripped off. opinions?

The model publish itself is intended to exclude shaders and sets but it publishes along a look set I believe, as per _this logic here:_ https://github.com/ynput/OpenPype/blob/7be7c8292aace23a6a1926985fdfede2bf26caa6/openpype/hosts/maya/plugins/publish/collect_look.py#L630-L655

Are you saying that _that_ would need to include the vray sets? if so, then that makes sense, yes. However, for me I was unable to quickly get that look set to be extracted. It could be that a certain plug-in needs to be enabled to ensure those get exported. (Even though looking at the code it would still exclude vray sets)

**Comment by BigRoy** [comment]
@antirotor as pointed out above - the look render sets applying still needs a fix ☝️

**Comment by antirotor** [comment]
> @antirotor as pointed out above - the model look render sets applying still needs a fix ☝️

I found the issue and it is puzzling. I am not sure if it was there for some time or it is new behaviour of Maya 2024, but it seems that when `cmds.ls(render_sets, type=render_set_types)` encounter type that isn't available, `ls()` will issue warning about unknown type but returns empty list.

I didn't test it in older versions of Maya, but this fix works for me in 2024 and VRay. I've also fixed that inventory action.

**Comment by jakubjezek001** [comment]
Task linked: [OP-7457 Maya: fix crashing model renderset collector](https://app.clickup.com/t/6658547/OP-7457)

**Review by LiborBatek — APPROVED** [review]
Works fine!

image

File: openpype/pipeline/actions.py [hunk 1]
```diff
@@ -100,6 +102,11 @@ def process(self, containers):
         """
         return True
 
+    @classmethod
+    def filepath_from_context(cls, context):
+        return get_representation_path_from_context(context)
+
+
```
Comment (BigRoy) [inline openpype/pipeline/actions.py:109]: Isn't this one empty line too many?
```suggestion
```

File: openpype/hosts/maya/plugins/publish/collect_look.py [hunk 2]
```diff
@@ -367,19 +367,34 @@ def collect(self, instance):
...
+                    "VRayObjectProperties",
+                ]
+
+            if cmds.pluginInfo("redshift4maya", query=True, loaded=True):
+                render_set_types += [
+                    "RedshiftObjectId",
+                    "RedshiftMeshParameters",
+                ]
+
             render_sets = cmds.ls(look_sets, type=render_set_types)
```
[the remark this answers is not in the document]
Comment (antirotor) [reply openpype/hosts/maya/plugins/publish/collect_look.py]: I've adapted it a little bit but other changes unrelated to this issue should go to other PR

File: openpype/pipeline/actions.py (hunk 1 above)
Comment (antirotor) [reply openpype/pipeline/actions.py:109]: > Isn't this one empty line too many?

shouldn't be, next function isn't part of the class `InventoryAction` so it should be separated by two lines

Comment (BigRoy) [reply openpype/pipeline/actions.py:109]: I thought github was showing me three lines somehow :/ sorry,

**Pull request merged**